Libplanet.Net/CandidateBranch.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Libplanet.Action;
+using Libplanet.Blockchain;
+using Libplanet.Blocks;
+
+namespace Libplanet.Net
+{
+    /// <summary>
+    /// A class to store an appendable <see cref="CandidateBranch{T}"/> for a
+    /// <see cref="BlockChain{T}"/>. <see cref="CandidateBranch{T}.Blocks"/> can be empty if
+    /// <see cref="CandidateBranch{T}"/> was created with non-continuous blocks.
+    /// </summary>
+    /// <typeparam name="T">An <see cref="IAction"/> type. It should match
+    /// to <see cref="Block{T}"/>'s type parameter.</typeparam>
```

Comment:
It is generally better to not allow creation of a `Branch` by throwing an exception rather than allowing a "special" type of branch (i.e. an empty branch) if invalid argument is passed on to a constructor. 😕

Reply:
I see, I wasn't sure whether throwing exception in constructor is expectable. 🤔 I replaced empty branch strategy to throw `CreateCandidateBranchException` and caught `Exception` where the `CanddiateBranch<T>` is created.